src/core/shape_inference/include/nms_shape_inference.hpp

```diff
@@ -8,14 +8,20 @@
 #include <openvino/op/non_max_suppression.hpp>
 #include <vector>
 
+#include "utils.hpp"
+
 using namespace ngraph;
 
 namespace ov {
 namespace op {
 namespace v9 {
 
 template <class T>
-void shape_infer(const NonMaxSuppression* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+void shape_infer(const NonMaxSuppression* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 bool static_output = false,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
```

Comment:
```suggestion
                 const std::map<size_t, ov::Tensor>& constant_data = {}) {
```

Reply:
@jane-intel Can we use new API?

Reply:
This is added to align with other shape inference, and there is currently only supported API that creates host tensor in GPU plugin side.

Reply:
Please find a ticket to do it simultaneously for all the ops: 95057